rfcs/NNNN-decorator-roadmap.md

```diff
@@ -0,0 +1,278 @@
+class El extends LitElement {
+  @property()
+  reactiveProp = "initial value";
+
+  @state()
+  secondProperty = 0;
+}
+```
+
+2. Incrementally make decorator usage _use site_ compatible by adding `accessor`
+
+```ts
+// tsconfig uses `experimentalDecorators: true`.
+class El extends LitElement {
+  @property()
+  accessor reactiveProp = "initial value";
+
+  @state()
+  accessor secondProperty = 0;
+}
+```
+
+3. Finally, change `experimentalDecorators` to `false` to complete the migration.
+
+## Implementation Considerations
+
+### Implementation Plan
+
+The plan is to make the Lit decorators work in both experimental decorator
+environments and standard decorator environments.
+
+#### I. Make Lit decorators hybrid (breaking)
+
+This stage adds support for the new decorators standard for both TypeScript and JavaScript developers,
+and ensures experimental decorators are semantically identical to standard decorators.
+
+##### Requirements
+
+- TypeScript ships decorator metadata (TypeScript 5.2, scheduled for 2023-08-22)
+
+##### Changes
+
+- Make Lit decorators "hybrid", working in either experimental or standard environments.
+- Non-core packages with decorators (@lit/labs) must follow a similar plan
+- Update `static elementProperties` to read from `[Symbol.metadata]`
+  - Standard decorators cannot call into the `static createProperty()` API, so they must place property options into `[Symbol.metadata]`. We should use that as the source-of-truth going forward.
+
+#### II. Deprecate experimental decorators
+
+##### Requirements
+
+- Stage I has landed.
+- Preferred: At least one browser has shipped decorators
+
+##### Changes
+
+- Deprecate the `noAccessor` property option.
+- Deprecate experimental decorators, `static createProperty()`, `static getPropertyDescriptor()`, etc. but _not_ `static properties`.
+  - Since there are no native implementations of decorators yet, we don't quite yet want to deprecate the main developer-facing API for property declaration.
+
+
+#### III. Remove experimental decorators (breaking)
+
+This stage requires decorators for creating properties. It is no longer usable in non-decorator-supporting environments without transpilation.
+
+##### Requirements
+
+- Stage II has landed
+- Native decorators are shipping in all major browsers.
+
+##### Changes
+
+- Remove previously deprecated APIs
+- Deprecate `static properties`
+- Re-export `@customElement()`, `@property()` and `@state()` from the main reactive-element, lit-element, and lit modules.
+  - This increases the core module size, which is paid for by removing the deprecated APIs.
+  - Other decorators are more optional and remain in their own modules.
+
+#### V. Cleanup (breaking)
+
+When all developers can use the standard decorators API we can remove the experimental decorators fully as well as `static properties`.
+
+##### Requirements
+
+- Stage IV has landed
+- Native decorators are shipping in all major browsers.
```

Comment:
```suggestion
- Native decorators have been shipping in all major browsers for some time.
```
or maybe we want to say for some specified time period/version like "for at least the last 2 major version" ?

Reply:
I like the last 2 major versions verbiage. It's more concretely actionable.

Reply:
```suggestion
- Native decorators are have been shipping in all major browsers for last 2 major versions.
```

Reply:
Ship it! :D